CopilotKit/packages/react-core/src/hooks/use-copilot-chat.ts

```diff
@@ -205,13 +217,34 @@ export function useCopilotChat({
   const latestRunChatCompletionFunc = useCallback(() => {
     return latestRunChatCompletion.current!();
   }, [latestRunChatCompletion]);
+  
+  const reset = useCallback(() => {
+    latestStopFunc();
+    setMessages([]);
+    setThreadId(null);
+    setRunId(null);
+    setCoagentStatesWithRef({});
```

Comment:
This used to be `setCoagentStates` but looks like that was removed in some of the newer changes. Just wanted to highlight it.